File: frontend/src/types/task.ts
export type TaskStatus = 'TODO' | 'IN_PROGRESS' | 'DONE';
export type TaskPriority = 'LOW' | 'MEDIUM' | 'HIGH';

export interface Task {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  dueDate: string | null;
  assignedTo: string | null;
  createdBy: string;
  aiSuggestions: string[];
  createdAt: string;
  updatedAt: string;
}

export interface TaskComment {
  id: string;
  taskId: string;
  userId: string;
  content: string;
  createdAt: string;
  user?: {
    fullName: string;
    email: string;
  };
}

export interface CreateTaskInput {
  title: string;
  description: string;
  priority: TaskPriority;
  dueDate?: string;
  assignedTo?: string;
}

export interface UpdateTaskInput extends Partial<CreateTaskInput> {
  status?: TaskStatus;
}

export interface TaskFilters {
  status?: TaskStatus;
  priority?: TaskPriority;
  assignedTo?: string;
  search?: string;
}

export interface TaskContextType {
  tasks: Task[];
  loading: boolean;
  error: string | null;
  createTask: (task: CreateTaskInput) => Promise<void>;
  updateTask: (id: string, task: UpdateTaskInput) => Promise<void>;
  deleteTask: (id: string) => Promise<void>;
  getAISuggestions: (taskDescription: string) => Promise<string[]>;
} 

File: task-ai-backend/models/task.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type TaskStatus string
type TaskPriority string

const (
	TaskStatusTodo       TaskStatus = "TODO"
	TaskStatusInProgress TaskStatus = "IN_PROGRESS"
	TaskStatusDone       TaskStatus = "DONE"

	TaskPriorityLow    TaskPriority = "LOW"
	TaskPriorityMedium TaskPriority = "MEDIUM"
	TaskPriorityHigh   TaskPriority = "HIGH"
)

type Task struct {
	ID            uuid.UUID    `json:"id" db:"id"`
	Title         string       `json:"title" db:"title"`
	Description   string       `json:"description" db:"description"`
	Status        TaskStatus   `json:"status" db:"status"`
	Priority      TaskPriority `json:"priority" db:"priority"`
	DueDate       *time.Time   `json:"dueDate,omitempty" db:"due_date"`
	AssignedTo    *string      `json:"assignedTo,omitempty" db:"assigned_to"`
	CreatedBy     string       `json:"createdBy" db:"created_by"`
	AIsuggestions []string     `json:"aiSuggestions,omitempty" db:"ai_suggestions"`
	CreatedAt     time.Time    `json:"createdAt" db:"created_at"`
	UpdatedAt     time.Time    `json:"updatedAt" db:"updated_at"`
}

type CreateTaskInput struct {
	Title       string       `json:"title"`
	Description string       `json:"description"`
	Priority    TaskPriority `json:"priority"`
	DueDate     *time.Time   `json:"dueDate,omitempty"`
	AssignedTo  *string      `json:"assignedTo,omitempty"`
}

type UpdateTaskInput struct {
	Title       *string       `json:"title,omitempty"`
	Description *string       `json:"description,omitempty"`
	Status      *TaskStatus   `json:"status,omitempty"`
	Priority    *TaskPriority `json:"priority,omitempty"`
	DueDate     *time.Time    `json:"dueDate,omitempty"`
	AssignedTo  *string       `json:"assignedTo,omitempty"`
}

type AITaskSuggestionInput struct {
	Description string `json:"description"`
}
